### PR TITLE
[ArrowStringArray] REF: extract/extractall column names

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

File: pandas/core/strings/accessor.py [hunk 1]
```diff
@@ -3036,13 +3038,31 @@ def _result_dtype(arr):
...
-        return list(rx.groupindex.keys()).pop()
-    except IndexError:
+def _get_single_group_name(regex: Pattern) -> Hashable:
+    if regex.groupindex:
+        return next(iter(regex.groupindex))
+    else:
         return None
 
 
+def _get_group_names(regex: Pattern) -> List[Hashable]:
```
Comment (datapythonista) [inline pandas/core/strings/accessor.py:3048]: This code has been working for a while, but wondering if it'd be useful to add a test for this function, seems trivial to add.

**Review by datapythonista — APPROVED** [review]
lgtm, nice clean up

**Comment by jreback** [comment]
thanks @simonjayhawkins 

wouldn't object to moar tests

**Comment by simonjayhawkins** [comment]
the column naming already seems to be extensively tested in the extract/extractall tests with named groups, unnamed groups, some unnamed groups, one group, two groups etc.

this is a precursor to #41372, same for extractall and then the dispatch to pyarrow kernels and perf opt of the object fallback. will add tests where there appear to be gaps. (but I don't think DataFrame column names or Index/Series names is one of those gaps)

**Pull request merged**